### PR TITLE
Redux and components: split ReduxProps from Props and type State

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
redux type expression-keypad
type redux props empty-keypad-button
redux type fraction-keypad
type mapstatetoprops keypad
type redux for two-page-keypad
use Redux state instead of any

Issue: LC-740

## Test plan:

### CODE REVIEW

File: packages/math-input/src/components/empty-keypad-button.tsx [hunk 1]
```diff
@@ -42,7 +44,7 @@ class EmptyKeypadButton extends React.Component<Props> {
     }
 }
 
-const mapStateToProps = (state) => {
+const mapStateToProps: (state: State) => ReduxProps = (state) => {
```
Comment (kevinbarabash) [inline packages/math-input/src/components/empty-keypad-button.tsx:47]: This level of type annotation isn't necessary.  The following should be enough:
```
const mapStateToProps = (state:State): ReduxProps => {
```
If that works, could you update the rest of the `mapStateToProps` definitions?

File: packages/math-input/src/components/fraction-keypad.tsx [hunk 2]
```diff
@@ -15,16 +15,20 @@ import Keypad from "./keypad";
...
+interface ReduxProps {
+    cursorContext?: CursorContext;
     dynamicJumpOut: boolean;
+}
+
+interface Props extends ReduxProps {
     roundTopLeft: boolean;
     roundTopRight: boolean;
-};
+}
```
Comment (kevinbarabash) [inline packages/math-input/src/components/fraction-keypad.tsx:31]: Thanks for splitting up these props.  Please be cautious with `interfaces` which multiple definitions of the same interface will get merged.  This is especially scary with things like `Node` which is a globally defined interface.

File: packages/math-input/src/components/keypad.tsx [hunk 3]
```diff
@@ -140,9 +141,9 @@ class Keypad extends React.Component<Props> {
     }
 }
 
-const mapStateToProps = (state) => {
+const mapStateToProps: (state: State) => ReduxProps = (state) => {
     return {
-        ...state.echoes,
+        echoes: state.echoes.echoes,
```
Comment (kevinbarabash) [inline packages/math-input/src/components/keypad.tsx:146]: It would be nice to get rid of the extra layer of indirection.  Not necessary for this PR.

**Review by kevinbarabash — COMMENTED** [review]
Thanks for splitting up the props.  It's much easier to follow. 💚

File: packages/math-input/src/components/fraction-keypad.tsx (hunk 2 above)
Comment (jeremywiebe) [reply packages/math-input/src/components/fraction-keypad.tsx:31]: I was surprised to see `interface` also, but that's how the docs suggest to do it: https://react-redux.js.org/using-react-redux/usage-with-typescript#inferring-the-connected-props-automatically

**Review by jeremywiebe — APPROVED** [review]
Nice! 💯

**Pull request merged**